tests/acceptance/features/webUILogin/login.feature

```diff
@@ -70,3 +70,16 @@ Feature: login users
     Then the user should be redirected to the login error page
     When the user exits the login error page
     Then the user should be redirected to the login page
+
+  Scenario: the user session of a deleted user should not be valid for newly created user of same name
+    Given these users have been created with default attributes:
+      | username |
+      | user1    |
+    And user "user1" has logged in using the webUI
+    And the user has browsed to the files page
+    And user "user1" has been deleted
+    And the user has reloaded the current page of the webUI
+    And the user has been redirected to the login error page
```

Comment:
I think I have some confusion in the given steps. for example in this case, `And the user has been redirected to the login error page` is in the given step, but it is actually the then step of the previous line i.e `the user has reloaded the current page of the WebUI`. What I think is, these two can be combined actually, something like `And the user has been redirected to the login error page after reloading the current page of the webui` Or we can have a new scenario where the user after being deleted reloads the current page in the `when` step and then its then step is `it should be redirected to the login error page` and in this scenario, we can use a single given step for that such as `And the deleted user has reloaded the current page of the WebUI` whose implementation can be the reload action and the assertion of being redirected to the login error page. But I am not sure if the present implementation is fine. Can you make me clear @phil-davis @individual-it @swoichha ?

Reply:
I think we have 2 tests here:
1. When the user is deleted And user reload the page Then the user should be redirected to new page
2. When the user is deleted And the user is recreated And user reload the page Then the user should be redirected to new page

The first test should pass, the second one should fail because of the bug

Will separating the test help to solve the confusion? And maybe we even don't need new steps for that

Reply:
@jasson99 You are right line 82 is the `Then` step of line 81. I wrote it in `Given` step because it is the precondition for this tests. I can combine these two into a single step as you have suggested but making a new scenario is not necessary for "where the user after being deleted reloads the current page" as there are already some test for that case.

Reply:
Or maybe I can just change them as @phil-davis suggested here https://github.com/owncloud/web/pull/4682#discussion_r569130670